test/VaultInventoryReporter.ts

```diff
@@ -0,0 +1,1225 @@
+                tokenAddress: mockERC721.address,
+                tokenId: id721,
+                tokenAmount: 0
+            },
+            {
+                itemType: ERC_1155_ITEM_TYPE,
+                tokenAddress: mockERC1155.address,
+                tokenId: id1155,
+                tokenAmount: amount1155
+            },
+            {
+                itemType: PUNKS_ITEM_TYPE,
+                tokenAddress: punks.address,
+                tokenId: idPunk,
+                tokenAmount: 0
+            }
+        ];
+    });
+
+    describe("Inventory Operations", () => {
+        describe("add", () => {
+
+            it("should revert if attempting to add zero items", async () => {
+                const { reporter, user, vault } = ctx;
+
+                await expect(
+                    reporter.connect(user).add(vault.address, [])
+                ).to.be.revertedWith("VIR_NoItems");
+            });
+
+            it("should revert if attempting to add more items than the maximum", async () => {
+                const { reporter, user, vault, mockERC20 } = ctx;
+
+                const item: Item = {
+                    itemType: ERC_20_ITEM_TYPE,
+                    tokenAddress: mockERC20.address,
+                    tokenId: 0,
+                    tokenAmount: amount20
+                };
+
+                // Create a vault of 51 items.
+                const items = (Array(51) as Item[]).fill(item, 0, 51);
+
+                await expect(
+                    reporter.connect(user).add(vault.address, items)
+                ).to.be.revertedWith("VIR_TooManyItems");
+            });
+
+            it("should revert if the caller is not the vault's owner or approved", async () => {
+                const { reporter, other, vault, mockERC20 } = ctx;
+
+                const item: Item = {
+                    itemType: ERC_20_ITEM_TYPE,
+                    tokenAddress: mockERC20.address,
+                    tokenId: 0,
+                    tokenAmount: amount20
+                };
+
+                const items = [item];
+
+                await expect(
+                    reporter.connect(other).add(vault.address, items)
+                ).to.be.revertedWith("VIR_NotApproved");
+            });
+
+            it("should revert if trying to add a token with address 0", async () => {
+                const { reporter, user, vault } = ctx;
+
+                const item: Item = {
+                    itemType: ERC_20_ITEM_TYPE,
+                    tokenAddress: ZERO_ADDRESS,
+                    tokenId: 0,
+                    tokenAmount: amount20
+                };
+
+                const items = [item];
+
+                await expect(
+                    reporter.connect(user).add(vault.address, items)
+                ).to.be.revertedWith("VIR_InvalidRegistration");
+            });
+
+
+            it("should revert if not enough ERC20 tokens are held by the vault", async () => {
```

Comment:
Not sure if you meant "should revert if not enough ERC20 tokens are held by the user"? Since the test is depositing into the vault. Same with next 3 tests.